mopidy_tidal/library.py

```diff
@@ -222,50 +222,55 @@ def browse(self, uri) -> list[Ref]:
-            return ref_models_mappers.create_tracks(
-                self._get_album_tracks(session, parts[2])
-            )
-
-        if nr_of_parts == 3 and parts[1] == "artist":
-            top_10_tracks = ref_models_mappers.create_tracks(
-                self._get_artist_top_tracks(session, parts[2])[:10]
-            )
-
-            albums = ref_models_mappers.create_albums(
-                self._get_artist_albums(session, parts[2])
-            )
-
-            return albums + top_10_tracks
-
-        if nr_of_parts == 3 and parts[1] == "playlist":
-            return ref_models_mappers.create_tracks(
-                self._get_playlist_tracks(session, parts[2])
-            )
-
-        if nr_of_parts == 3 and parts[1] == "mood":
-            return ref_models_mappers.create_playlists(
-                self._get_mood_items(session, parts[2])
-            )
-
-        if nr_of_parts == 3 and parts[1] == "genre":
-            return ref_models_mappers.create_playlists(
-                self._get_genre_items(session, parts[2])
-            )
-
-        if nr_of_parts == 3 and parts[1] == "mix":
-            return ref_models_mappers.create_tracks(
-                self._get_mix_tracks(session, parts[2])
-            )
-
-        if nr_of_parts == 3 and parts[1] == "page":
-            return ref_models_mappers.create_mixed_directory(session.page.get(parts[2]))
-
-        logger.debug("Unknown uri for browse request: %s", uri)
-        return []
+        try:
+            _, type, id = uri.split(":")
+
+            if type == "album":
+                return ref_models_mappers.create_tracks(
+                    self._get_album_tracks(session, id)
+                )
+
+            elif type == "artist":
+                top_10_tracks = ref_models_mappers.create_tracks(
+                    self._get_artist_top_tracks(session, id)[:10]
+                )
+
+                albums = ref_models_mappers.create_albums(
+                    self._get_artist_albums(session, id)
+                )
+
+                return albums + top_10_tracks
+
+            elif type == "playlist":
+                return ref_models_mappers.create_tracks(
+                    self._get_playlist_tracks(session, id)
+                )
+
+            elif type == "mood":
+                return ref_models_mappers.create_playlists(
+                    self._get_mood_items(session, id)
+                )
+
+            elif type == "genre":
+                return ref_models_mappers.create_playlists(
+                    self._get_genre_items(session, id)
+                )
+
+            elif type == "mix":
+                return ref_models_mappers.create_tracks(
+                    self._get_mix_tracks(session, id)
+                )
+
+            elif type == "page":
+                return ref_models_mappers.create_mixed_directory(session.page.get(id))
+            else:
+                return []
+        except ValueError:
```

Comment:
Double check this exception type is still used after my latest additions to tidalapi